chain-extensions/pallet-assets/src/lib.rs

```diff
@@ -111,8 +123,15 @@ where
                 let base_weight = <T as pallet_assets::Config>::WeightInfo::create();
                 env.charge_weight(base_weight)?;
 
+                let raw_origin = match origin {
+                    Origin::Caller => {
+                        return Ok(RetVal::Converging(Outcome::OriginCannotBeCaller as u32))
+                    }
+                    Origin::Address => RawOrigin::Signed(env.ext().address().clone()),
+                };
```

Comment:
Please write a macro for this - it's repeated many times and given how sensitive this is, better to have it implemented exactly _once_.